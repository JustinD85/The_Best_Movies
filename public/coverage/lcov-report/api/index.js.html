<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for api/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">api</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>28/28</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>23/23</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import apiKey from './movie_api_key'
&nbsp;
// const dbUrl = 'http://localhost:3000/api/'
const dbUrl = 'https://falsemotive.io/api/'
const moviesUrl = 'https://api.themoviedb.org/3/movie/'
export const options = (method, data) =&gt; ({
  method,
  headers: {
    'Content-Type': 'application/json',
  },
  body: JSON.stringify(data)
})
&nbsp;
export const postData = async (path, data) =&gt; {
  const response = await fetch(`${dbUrl}${path}`, options('POST', data))
  if (!response.ok) throw new Error(`Could not post`)
  const unfilteredData = await response.json()
  return unfilteredData.data
}
&nbsp;
export const deleteData = async (path, data) =&gt; {
  const response = await fetch(`${dbUrl}${path}`, options('DELETE', data))
  if (!response.ok) throw new Error(`Could not delete`)
  const unfilteredData = await response.json()
  return unfilteredData.results
}
&nbsp;
export const getData = async (path) =&gt; {
  const response = await fetch(`${dbUrl}${path}`)
  if (!response.ok) throw new Error(`Could not get Data`)
  const unfilteredData = await response.json()
  return unfilteredData.data
}
&nbsp;
export const getMovies = async (path) =&gt; {
  const response = await fetch(`${moviesUrl}${path}${apiKey}`)
  if (!response.ok) throw new Error('Could not get Movie')
  const unfilteredMovies = await response.json()
  return unfilteredMovies.results
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 06 2019 09:52:14 GMT-0700 (Mountain Standard Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
