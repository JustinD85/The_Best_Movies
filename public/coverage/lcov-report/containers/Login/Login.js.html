<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for containers/Login/Login.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">containers/Login</a> Login.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">85.71% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>30/35</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>12/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">84.85% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>28/33</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react'
import { NavLink, Redirect, Link } from 'react-router-dom'
import { postData, getData } from '../../api'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { setMovies, setUser, setFavorites, setLoginPrompt, setPopup } from '../../actions'
import './Login.scss'
&nbsp;
export class Login extends Component {
  constructor() {
    super()
    this.state = {
      email: '',
      password: '',
      error: '',
      canLogin: false,
    }
  }
&nbsp;
  handleChange = <span class="fstat-no" title="function not covered" >(e</span>) =&gt; {
    const { name, value } = <span class="cstat-no" title="statement not covered" >e.target;</span>
<span class="cstat-no" title="statement not covered" >    this.setState({</span>
      [name]: value,
    })
  }
&nbsp;
  setUser = async () =&gt; {
    const { password, email } = this.state
    const data = await postData('users', { password, email })
    const { name, id } = data
    localStorage.setItem('users', JSON.stringify({ name, id }))
    this.props.setUser({ name, id })
    return id;
  }
&nbsp;
  setUserFavorites = async (userID) =&gt; {
    const retrieveFavPath = `users/${userID}/favorites`
    const favorites = await getData(retrieveFavPath)
    this.props.setFavorites(favorites, userID)
  }
&nbsp;
  handleSubmit = async (e) =&gt; {
    e.preventDefault();
    try {
      const userID = await this.setUser()
      this.setUserFavorites(userID)
      this.setState({ canLogin: true })
      this.props.setPopup(false)
    } catch (error) {
<span class="cstat-no" title="statement not covered" >      this.setState({</span>
        error: 'That email or password does not exist'
      })
&nbsp;
    }
  }
&nbsp;
  componentDidMount = () =&gt; {
    this.props.setPopup(true)
    this.props.setLoginPrompt(false)
  }
&nbsp;
  render() {
    const { canLogin, error } = this.state
&nbsp;
    return (
      canLogin ? &lt;Redirect to='/' /&gt; :
        &lt;form className='Login' onSubmit={this.handleSubmit}&gt;
          &lt;h1&gt;Movie Tracker&lt;/h1&gt;
          &lt;h3&gt;Welcome back!&lt;/h3&gt;
          &lt;input
            className='user-input'
            placeholder='Email'
            required type='email'
            onChange={this.handleChange}
            name='email' /&gt;
          &lt;input
            className='user-input'
            placeholder='Password'
            required type='password'
            onChange={this.handleChange}
            name='password' /&gt;
          {error &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >&lt;h3&gt;{error}&lt;/h3&gt;}</span>
          &lt;button className='sign-in link'&gt;Sign In&lt;/button&gt;
          &lt;p&gt;New to Movie Tracker?&lt;/p&gt;
          &lt;NavLink className='create-user link' to='/create-user'&gt;Create Account&lt;/NavLink&gt;
          &lt;Link className='back-btn' to='/' onClick={() =&gt; this.props.setPopup(false)}&gt;Go Back&lt;/Link&gt;
        &lt;/form&gt;
    )
  }
}
&nbsp;
export const mapStateToProps = (state) =&gt; ({
  movies: state.movies,
})
&nbsp;
export const mapDispatchToProps = (dispatch) =&gt; ({
  setMovies: (movies) =&gt; dispatch(setMovies(movies)),
  setUser: <span class="fstat-no" title="function not covered" >(u</span>ser) =&gt; <span class="cstat-no" title="statement not covered" >dispatch(setUser(user)),</span>
  setFavorites: <span class="fstat-no" title="function not covered" >(f</span>avorites, user_id) =&gt; <span class="cstat-no" title="statement not covered" >dispatch(setFavorites(favorites, user_id)),</span>
  setLoginPrompt: (bool) =&gt; dispatch(setLoginPrompt(bool)),
  setPopup: (bool) =&gt; dispatch(setPopup(bool)),
})
&nbsp;
export default connect(mapStateToProps, mapDispatchToProps)(Login)
&nbsp;
Login.propTypes = {
  setLoginPrompt: PropTypes.func,
  setPopup: PropTypes.func,
  setFavorites: PropTypes.func,
  setUser: PropTypes.func,
}</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 06 2019 09:52:14 GMT-0700 (Mountain Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
