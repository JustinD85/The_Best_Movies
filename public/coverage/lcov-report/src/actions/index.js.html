<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/actions/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/actions</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// Movie key
&nbsp;
export const setFilter = (filter) =&gt; ({
    type: 'SET_FILTER',
    filter,
})
&nbsp;
export const toggleFavorite = <span class="fstat-no" title="function not covered" >(i</span>d)=&gt;(<span class="cstat-no" title="statement not covered" >{</span>
    type:'TOGGLE_FAVORITE',
    id,
})
&nbsp;
export const setMovies = <span class="fstat-no" title="function not covered" >(m</span>ovies) =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  type: 'SET_MOVIES',
  movies,
})
&nbsp;
// Users key 
&nbsp;
export const addUser = <span class="fstat-no" title="function not covered" >(n</span>ame, password, email) =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
    type: 'ADD_USER',
    name,
    password,
    email,
})
&nbsp;
export const setUsers = <span class="fstat-no" title="function not covered" >(u</span>sers) =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
    type: 'SET_USERS',
    users,
})</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 30 2019 20:39:56 GMT-0700 (Mountain Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
