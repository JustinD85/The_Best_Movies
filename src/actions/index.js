export const setFilter = (filter) => ({
    type: 'SET_FILTER',
    filter,
})

export const toggleFavorite = (id)=>({
    type:'TOGGLE_FAVORITE',
    id,
})
