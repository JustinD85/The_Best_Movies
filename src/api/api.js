// const url = 'http://localhost:3000/api/'
const url = 'https://falsemotive.io/api/'

export const postData = async (options = '', user) => {
  const response = await fetch(`${url}${options}`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(user)
  })

  if (response.ok) {
    return response;
  } else {
    throw new Error(`Error posting data: ${response.status}`);
  }
}

export const fetchData = async (options) => {
  const response = await fetch(`${url}${options}`)
  if (response.ok) {
    return await response.json()
  } else {
    throw new Error('Error fetching data')
  }
}