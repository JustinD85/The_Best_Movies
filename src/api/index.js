import apiKey from './movie_api_key'

// const dbUrl = 'http://localhost:3000/api/'
const dbUrl = 'https://falsemotive.io/api/'
const moviesUrl = 'https://api.themoviedb.org/3/movie/'
export const options = (method, data) => ({
  method,
  headers: {
    'Content-Type': 'application/json',
  },
  body: JSON.stringify(data)
})

export const postData = async (path = '', data) => {
  const response = await fetch(`${dbUrl}${path}`, options('POST', data))
  if (!response.ok) throw new Error(`Could not post`)
  const unfilteredData = await response.json()
  return unfilteredData.data
}

export const deleteData = async (path = '', data) => {
  const response = await fetch(`${dbUrl}${path}`, options('DELETE', data))
  if (!response.ok) throw new Error(`Could not delete`)
  const unfilteredData = await response.json()
  return unfilteredData.results
}

export const getData = async (path) => {
  const response = await fetch(`${dbUrl}${path}`)
  if (!response.ok) throw new Error(`${response.statusText}`)
  const unfilteredData = await response.json()
  return unfilteredData.data
}

export const getMovies = async (path) => {
  const response = await fetch(`${moviesUrl}${path}${apiKey}`)
  if (!response.ok) throw new Error(`${response.statusText}`)
  const unfilteredMovies = await response.json()
  return unfilteredMovies.results
}
