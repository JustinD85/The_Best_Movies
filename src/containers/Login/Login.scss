.Login {
  height: 62%;
  width: 68%;
  z-index: 1;
  position: absolute;
  left: 22%;
  top: 10%;
  border-radius: 15px;
  background: rgba(0, 0, 0, 0.9);
  color: #ECEBEE;
  padding: 90px 30px 0px 30px;

  .back-btn {
    text-decoration: none;
    color: #ECEBEE;

    &:hover {
      color: #0CD0FC;

    }
  }

  p,
  h3 {
    color: whitesmoke;
  }

  h1 {
    font-family: 'Pacifico', sans-serif;
    align-self: center;
    grid-area: title;
    font-size: 4rem;
    color: #0CD0FC;
  }

  .link {
    color: #000;
    margin: 10px auto;
    width: 36%;
    background: whitesmoke;
    border-radius: 25px;
    display: grid;
    justify-content: center;
    align-items: center;
    text-decoration: none
  }

  .user-input {
    font-size: 14px;
    border-radius: 5px;
    padding: 6px;
    margin: 10px;
  }

}