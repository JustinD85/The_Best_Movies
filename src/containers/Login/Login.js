import React, { Component } from 'react'
import { NavLink } from 'react-router-dom'
import { postData } from '../../api/api'
import './Login.scss'

class Login extends Component {
  constructor() {
    super()
    this.state = {
      email: '',
      password: '',
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target;

    this.setState({
      [name]: value,
    })
  }

  // handleSubmit = async (e) => {
  //   e.preventDefault();
  //   const { password, email } = this.state
  //   const user = { password, email }

  //   try {
  //     const result = await postData('users', user)
  //     console.log(result)
  //   } catch (error) {
  //     this.setState({
  //       error: error.message
  //     })
  //   }
  // }

  render() {
    return (
      <div className='Login'>
        <h1>Movie Tracker</h1>
        <h3>Welcome back!</h3>
        <input
          className='user-input'
          placeholder='Email'
          required type='email'
          onChange={this.handleChange}
          name='email' />
        <input
          className='user-input'
          placeholder='Password'
          required type='password'
          onChange={this.handleChange}
          name='password' />
        <NavLink
          className='sign-in link'
          onClick={this.handleSubmit}
          to='/'>Sign In</NavLink>
        <p>New to Movie Tracker?</p>
        <NavLink className='create-user link' to='/create-user'>Create Account</NavLink>
      </div> 
    )
  }
}

export default Login;