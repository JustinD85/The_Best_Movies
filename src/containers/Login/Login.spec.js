import React from 'react'
import { shallow } from 'enzyme'
import { postData, getData } from '../../api'
import { setMovies, setUser, setFavorites, setShouldPromptLogin, setPopup } from '../../actions'
import { Login, mapDispatchToProps, mapStateToProps } from './Login'
jest.mock('../../api')

let wrapper
let setUserMock = jest.fn()
let setFavoritesMock = jest.fn()
let setPopupMock = jest.fn()
let setShouldPromptLoginMock = jest.fn()
let setMoviesMock = jest.fn()
window.fetch = jest.fn()
window.getData = jest.fn()

describe('Login', () => {

  beforeEach(() => {

    wrapper = shallow(<Login
      setUser={setUserMock}
      setFavorites={setFavoritesMock}
      setPopup={setPopupMock}
      setShouldPromptLogin={setShouldPromptLoginMock}
      setMovies={setMovies}
    />)
    postData.mockReturnValue({name:'johnny',id:1})
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call setPopupMock when link is clicked', () => {
    wrapper.find('Link').simulate('click')

    expect(setPopupMock).toHaveBeenCalled()
  })
  it('should call setFavoriteMock when link is clicked', async () => {
    wrapper.setState({ password: 'f', email: 'me@me.com' })
    await wrapper.instance().handleSubmit({ preventDefault: () => '' })

    await expect(setFavoritesMock).toHaveBeenCalled()
  })
  it('should set login to true when link is clicked',async () => {
    await wrapper.instance().handleSubmit({ preventDefault: () => '' })

    await expect(wrapper.state('canLogin')).toEqual(true)
  })


  describe('mapStateToProps', () => {
    let mockState = {
      movies: { movieOne: {} },
      eggs: 'scrambled',
      toast: 'toasted'
    }
    let expected = {
      movies : mockState.movies
    }

    let result = mapStateToProps(mockState)
    expect(result).toEqual(expected)

  })

  describe('mapDispatchToProps', () => {
    let mockDispatch = jest.fn()
    let props = mapDispatchToProps(mockDispatch)

    it('should call dispatch when setmovies is called ', () => {
      const expected = setMovies({ movies: 'yaa' })
      props.setMovies({ movies: 'yaa' })
      expect(mockDispatch).toHaveBeenCalledWith(expected)
    })

    it('should call dispatch setShouldPromptLogin is called ', () => {
      const expected = setShouldPromptLogin(true)
      props.setShouldPromptLogin(true)
      expect(mockDispatch).toHaveBeenCalledWith(expected)
    })

    it('should call dispatch setPopup is called ', () => {
      const expected = setPopup(false)
      props.setPopup(false)
      expect(mockDispatch).toHaveBeenCalledWith(expected)
    })

    it('should call dispatch setPopup is called ', () => {
      const expected = setPopup(true)
      props.setPopup(true)
      expect(mockDispatch).toHaveBeenCalledWith(expected)
    })

  })
})
