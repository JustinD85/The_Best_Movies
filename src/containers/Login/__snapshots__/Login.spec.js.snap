// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
<form
  className="Login"
  onSubmit={[Function]}
>
  <h1>
    Movie Tracker
  </h1>
  <h3>
    Welcome back!
  </h3>
  <input
    className="user-input"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    required={true}
    type="email"
  />
  <input
    className="user-input"
    name="password"
    onChange={[Function]}
    placeholder="Password"
    required={true}
    type="password"
  />
  <button
    className="sign-in link"
  >
    Sign In
  </button>
  <p>
    New to Movie Tracker?
  </p>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="create-user link"
    to="/create-user"
  >
    Create Account
  </NavLink>
  <Link
    className="back-btn"
    onClick={[Function]}
    replace={false}
    to="/"
  >
    Go Back
  </Link>
</form>
`;
